tests-integration/extensions_test.go

```diff
@@ -721,7 +719,28 @@ func TestTypingMultiplePoller(t *testing.T) {
 		},
 	})
 
-	// Get the response from v3
+	// Queue another response for Bob with Charlie typing.
+	// Since Alice's poller is in charge of handling typing notifications, this shouldn't
+	// show up on future responses.
+	v2.queueResponse(bobToken, sync2.SyncResponse{
+		Rooms: sync2.SyncRoomsResponse{
+			Join: map[string]sync2.SyncV2JoinResponse{
+				roomA: {
+					State: sync2.EventsResponse{
+						Events: roomState,
+					},
+					Timeline: sync2.TimelineResponse{
+						Events: []json.RawMessage{joinEv},
+					},
+					Ephemeral: sync2.EventsResponse{
+						Events: []json.RawMessage{json.RawMessage(`{"type":"m.typing","content":{"user_ids":["@charlie:localhost"]}}`)},
+					},
+				},
+			},
+		},
+	})
+
```

Comment:
Need to `v2.waitUntilEmpty(t, aliceToken)` and `bobToken` to make sure we do the request after the proxy has processed the v2 response.